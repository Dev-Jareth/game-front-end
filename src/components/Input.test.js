import React from 'react';
import Renderer from 'react-test-renderer';
import {Input} from './';

describe('Input Field',()=>{
  it('renders', () => {
    expect(Renderer.create(<Input />)).toMatchSnapshot();
  });
  it('has a placeholder',()=>{
    expect(Renderer.create(<Input placeholder="placeholder" />)).toMatchSnapshot();
  });
  it('has a value',()=>{
    expect(Renderer.create(<Input value="value" />)).toMatchSnapshot();
  });
  it('has an FA icon',()=>{
    expect(Renderer.create(<Input icon="gear" />)).toMatchSnapshot();
  });
  it('accepts an onUpdate function',()=>{
    expect(Renderer.create(<Input onUpdate= {val=>{console.log(val)}} />)).toMatchSnapshot();
  });
  it('rejects onUpdate non-function',()=>{
    fail('not implemented')
  })
  it('can error',()=>{
    expect(Renderer.create(<Input error />)).toMatchSnapshot();
    expect(Renderer.create(<Input error={true} />)).toMatchSnapshot();
  });
  it('can success',()=>{
    expect(Renderer.create(<Input success />)).toMatchSnapshot();
    expect(Renderer.create(<Input success={true} />)).toMatchSnapshot();
  });
  it('can disable',()=>{
    expect(Renderer.create(<Input disabled />)).toMatchSnapshot();
    expect(Renderer.create(<Input disabled={true} />)).toMatchSnapshot();
  })
  it('can have type',()=>{
    expect(Renderer.create(<Input type="text" />)).toMatchSnapshot();
    expect(Renderer.create(<Input type="password" />)).toMatchSnapshot();
  })

})
