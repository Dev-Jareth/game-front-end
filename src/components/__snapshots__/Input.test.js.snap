// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field accepts an onUpdate function 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can disable 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null disabled"
    disabled={true}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can disable 2`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null disabled"
    disabled={true}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can error 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control error null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can error 2`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control error null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can have type 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can have type 2`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="password"
    value=""
  />
</div>
`;

exports[`Input Field can success 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null success null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field can success 2`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null success null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field has a placeholder 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder="placeholder"
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field has a value 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value="value"
  />
</div>
`;

exports[`Input Field has an FA icon 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
  <label
    className="input-icon fa fa-gear"
    htmlFor="login-name"
  />
</div>
`;

exports[`Input Field rejects onUpdate non-function 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`Input Field renders 1`] = `
<div
  className="input-group"
>
  <input
    autoFocus={undefined}
    className="input-control null null null"
    disabled={undefined}
    id="login-name"
    maxLength={undefined}
    onChange={[Function]}
    placeholder={undefined}
    readOnly={undefined}
    required={undefined}
    type="text"
    value=""
  />
</div>
`;
